cuml/linear_regression_demo.ipynb

```diff
@@ -5,191 +5,275 @@
-    "X_train, X_test, y_train, y_test = load_data(nrows,ncols)\n",
-    "print('training data',X_train.shape)\n",
-    "print('training label',y_train.shape)\n",
-    "print('testing data',X_test.shape)\n",
-    "print('testing label',y_test.shape)\n",
-    "print('label',y_test.shape)"
+    "### Fit"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 5,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 5min 20s, sys: 3min 57s, total: 9min 18s\n",
+      "Wall time: 27.2 s\n"
+     ]
+    }
+   ],
    "source": [
     "%%time\n",
-    "# use the sklearn linear regression model to fit the training dataset \n",
-    "skols = skLinearRegression(fit_intercept=True,\n",
-    "                  normalize=True)\n",
+    "skols = skLR(fit_intercept=True,\n",
+    "             normalize=True,\n",
+    "             n_jobs=-1)\n",
+    "\n",
     "skols.fit(X_train, y_train)"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Evaluate"
+   ]
+  },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 6,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 1.48 s, sys: 600 ms, total: 2.08 s\n",
+      "Wall time: 209 ms\n"
+     ]
+    }
+   ],
    "source": [
     "%%time\n",
-    "# calculate the mean squared error of the sklearn linear regression model on the testing dataset\n",
     "sk_predict = skols.predict(X_test)\n",
+    "\n",
     "error_sk = mean_squared_error(y_test,sk_predict)"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "%%time\n",
-    "# convert the pandas dataframe to cudf format\n",
-    "X_cudf = cudf.DataFrame.from_pandas(X_train)\n",
-    "X_cudf_test = cudf.DataFrame.from_pandas(X_test)\n",
-    "y_cudf = y_train.values\n",
-    "y_cudf = y_cudf[:,0]\n",
-    "y_cudf = cudf.Series(y_cudf)"
+    "## cuML Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Fit"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
+   "execution_count": 10,
    "metadata": {},
-   "outputs": [],
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "CPU times: user 940 ms, sys: 400 ms, total: 1.34 s\n",
+      "Wall time: 1.35 s\n"
+     ]
+    }
+   ],
    "source": [
     "%%time\n",
-    "# run the cuml linear regression model to fit the training dataset \n",
-    "cuols = cuLinearRegression(fit_intercept=True,\n",
-    "                  normalize=True,\n",
-    "                  algorithm='eig')\n",
+    "cuols = cuLR(fit_intercept=True,\n",
```

Comment:
We have pretty inconsistent naming conventions for these objects, varying from camel case "cuOLS" to underscores to nothing... Don't need to fix this globally today, but doing an underscore here "cu_ols" would be more consistent with the rest of the notebook.

Reply:
That’s a reasonable. I can scrape through and fix them.